core/trino-main/src/main/java/io/trino/server/security/oauth2/OidcDiscovery.java

```diff
@@ -115,7 +114,7 @@ private OAuth2ServerConfig readConfiguration(String body)
             else {
                 userinfoEndpoint = Optional.empty();
             }
-            Optional<URI> endSessionEndpoint = Optional.of(getRequiredField("end_session_endpoint", metadata.getEndSessionEndpointURI(), END_SESSION_URL, Optional.empty()));
+            Optional<URI> endSessionEndpoint = Optional.ofNullable(metadata.getEndSessionEndpointURI());
```

Comment:
I think there are two issues here:
* `end_session_endpoint` is required here and you'd like it to be optional
* `end_session_endpoint` cannot be overridden by setting `io.trino.server.security.oauth2.StaticOAuth2ServerConfig#endSessionUrl`
How about
```java
Optional<URI> endSessionEndpoint = getOptionalField("end_session_endpoint", Optional.ofNullable(metadata.getEndSessionEndpointURI()).map(URI::toString), END_SESSION_URL, endSessionUrl);
```
where `endSessionUrl` comes from `io.trino.server.security.oauth2.StaticOAuth2ServerConfig#endSessionUrl`

Reply:
But do we need to introduce a config `endSessionUrl` bcoz the code mentions 
> TODO Left for backward compatibility, remove after the next release/a couple of releases. 

this statement was for the remaining config properties.

Reply:
You're right